src/lib.rs

```diff
@@ -110,14 +111,45 @@ impl VadSession {
         Self::new_from_bytes(model_bytes, config)
     }
 
+    /// Pass in some audio to the VAD and return a list of any speech transitions that happened
+    /// during the segment.
+    pub fn process(&mut self, audio_frame: &[f32]) -> Result<Vec<VadTransition>> {
+        const VAD_BUFFER_MS: usize = 30; // TODO This should be configurable
+        self.session_audio.extend_from_slice(audio_frame);
+
+        let vad_segment_length = VAD_BUFFER_MS * self.config.sample_rate / 1000;
+        let num_chunks = self.session_audio.len() / vad_segment_length;
+        let start_chunk = self.processed_samples;
+
+        let mut transitions = vec![];
+
+        for i in start_chunk..num_chunks {
+            let start_idx = i * vad_segment_length;
+            // we might not be getting audio chunks in perfect multiples of 30ms, so let the
```

Comment:
I was wondering if we still wanted to do this, but given there's no EOF to let us know when a packet is last this does also give us an advantage of processing all the audio and not risking dropping 20ms or so at the end because it's a remainder.

Reply:
(20ms shouldn't be possible - the last frame i.e. the remainder will be longer than 30ms, but shorter than 60ms)